target_postgres/__init__.py

```diff
@@ -347,16 +434,41 @@ def flush_records(stream, records_to_load, row_count, db_sync, temp_dir=None):
 
     size_bytes = 0
     csv_fd, csv_file = mkstemp(suffix='.csv', prefix=f'{stream}_', dir=temp_dir)
+    batch_bookmarks = []
+    batch_summary = {
+        "success": 0,
+        "fail": 0,
+        "existing": 0,
+        "updated": 0
+    }
     with open(csv_fd, 'w+b') as f:
         for record in records_to_load.values():
             csv_line = db_sync.record_to_csv_line(record)
+            batch_bookmarks.append({
+                "hash": build_record_hash(record),
+                "success": True,
+                "external_id": record.get('cid',""),
+                "reason": ""
+            })
             f.write(bytes(csv_line + '\n', 'UTF-8'))
 
     size_bytes = os.path.getsize(csv_file)
-    db_sync.load_csv(csv_file, row_count, size_bytes)
+    try:
+        inserted_lines, updated_lines = db_sync.load_csv(csv_file, row_count, size_bytes)
+        batch_summary["success"] += inserted_lines
+        batch_summary["updated"] += updated_lines
+    except Exception as e:
+        for batch_bookmark in batch_bookmarks:
+            batch_bookmark["reason"] = str(e)
```

Comment:
I think this should actually be `"error"` – I didn't give you a good example for that, sorry

Reply:
Fixed